### ISSUE TITLE
serialEvent not handled

### ISSUE THREAD

**Issue by asharect** [feilipu/Arduino_FreeRTOS_Library#7]
I used arduino example "SerialEvent", it was fine. But when I include Arduino_FreeRTOS, the serialEvent didn't work any more,  how to fix it? rewrite avr ISR? but I noticed that there's no configMAX_SYSCALL_INTERRUPT_PRIORITY option in freertosconfig.h, it seems that the uart interupt always is blocked by task, even idle task.

**Comment by feilipu** [comment]
I have tried to replicate the issue you describe, but I can't.
See below screenshot.
![screenshot from 2016-08-30 19-55-07](https://cloud.githubusercontent.com/assets/3955592/18084908/47c3c1ae-6eec-11e6-8b19-030c4ecf8b69.png)

The AVR ATmega is a very simple MCU. There is no interrupt prioritisation system (except intrinsic ordering). The configMAX_SYSCALL_INTERRUPT_PRIORITY option is not relevant to an ATmega port.

Perhaps you can reconfirm the issue?

**Comment by joe31093** [comment]
I see this is closed but I have the code running on a Teensy 3.6.  I have two tasks running.  Both are at idle + 1 priority.  One task toggles the LED on Pin 13 and the other is waiting for a semaphore give when the SerialEvent() is called.  I have a print in the serial event and it is not printing anything out.  I tested it the same way in the code you have above and it works fine.  Looks like running the scheduler affects SerialEvent.

**Comment by joe31093** [comment]
I am not sure if this helps.  I recently installed 8.2.3 (greiman) on my Teensy 3.6.  The serial event is not being call.  I have two tasks running.  A heartbeat task which turns on off the LEDs.  Another task waiting on a sem give in the serial event function so in can process the text.  I have a print statement in the serial event and it is not being printed.  I ran a similar example of the code you have above and the serial event worked fine.  I am guessing running the scheduler affected the system interrupts.
